Add support for different recipes (#10641)
RasaHQ/rasa: docs/docs/graph-recipe.mdx

---
id: graph-recipe
sidebar_label: Graph Recipe
title: Graph Recipe
description: Learn about Graph Recipe for Rasa Open Source.
abstract: Graph recipes provide a more fine tuned configuration for your executable graphs.
---

> We now support graph recipes in addition to the default recipe. What graph recipes provide is more powerful control over how execution graph schemas are built.

Reviewer: ```suggestion
We now support graph recipes in addition to the default recipe. Graph recipes provide more granular control over how execution graph schemas are built.
```

If you have feedback (positive or negative) please share it with us on the [Rasa Forum](https://forum.rasa.com).

:::

## Differences with Default Recipe

There are some differences between the default recipe and the new graph recipe. Main differences are:

- Default recipe is named `default.v1` in the config file whereas graph recipes are named `graph.v1`.
- Default recipes provide an easy to use recipe structure whereas graph recipes are more advanced and powerful.
- Default recipes are very opinionated and provide various defaults whereas graph recipes are more explicit.
- Default recipes can auto-configure themselves and dump the defaults used to the file if some sections in `config.yml` are missing, whereas graph recipes do none of this and assume what you see is what you get. There are no surprises with graph recipes.
- Default recipe divides graph configuration into mainly two parts: `pipeline` and `policies`. These can also be described as NLU and core (dialogue management) parts. For graph recipe on the other hand, the separation is between training (ie. `train_schema`) and prediction (ie. `predict_schema`).

:::tip Starting from scratch?

> If you don't know which recipe to choose, use the default recipe to bootstrap your project fast. If later you find that you need more powerful finetuning options, you can always change your recipe to be a graph recipe.

Reviewer: "finetuning" has a double meaning here 😁

Author: Double speak? How? :)

Reviewer: We use it to mean incremental training

Author: Oh yes, I see what you mean. I can change "finetuning" to "configuration" here?

abstract: Graph recipes provide a more fine tuned configuration for your executable graphs.
---

:::tip Default Recipe or Graph Recipe?

> You will probably only need graph recipes if you're running ML experiments or ablation studies on an existing model. We recommend starting with the default recipe and for many applications that will be all that's needed.

Reviewer: "ablation studies" feels a bit specific? (I had to google it 😁 )

Author: I've used the exact wording from @rctatman actually and I had to google it too :D I think it is good because it has the effect of warning against using graph recipes if the user is not too advanced.

:::tip Starting from scratch?

> If you don't know which recipe to choose, use the default recipe to bootstrap your project fast. If later you find that you need more powerful fine-tuning options, you can always change your recipe to be a graph recipe.

Reviewer: ```suggestion
If you don't know which recipe to choose, use the default recipe to bootstrap your project fast. If later you find that you need more fine-grained control, you can always change your recipe to be a graph recipe.
```
